### PR TITLE
Refactor `/patients/sidebar/action-sidebar.js` tests for json api

### PR DESCRIPTION
**Pull request by nmajor25**
Shortcut Story ID: [sc-53992]

### CODE REVIEW

File: test/support/api/teams.js [hunk 1]
```diff
@@ -50,6 +50,7 @@ export function getTeams({ attributes, relationships, meta } = {}, { depth = 0 }
 // Exporting only teams needed for testing variance
 export const teamCoordinator = getTeamResource({ id: '11111' });
 export const teamNurse = getTeamResource({ id: '22222' });
+export const teamPhysician = getTeamResource({ id: '44444' });
```
Comment (paulfalgout) [inline test/support/api/teams.js]: Can we change the tests to use `teamOther` I don't think should ever need more than 3.

File: test/integration/patients/sidebar/action-sidebar.js [hunk 2]
```diff
@@ -1601,51 +1602,54 @@ context('action sidebar', function() {
...
             sharing: 'disabled',
             details: '',
             duration: 0,
           },
           relationships: {
-            owner: { data: { id: '11111', type: 'clinicians' } },
-            state: { data: { id: '22222' } },
-            form: { data: { id: '11111' } },
-            files: { data: [{ id: '1' }] },
-            program_action: { data: { id: '1' } },
```
Comment (paulfalgout) [inline test/integration/patients/sidebar/action-sidebar.js:1638]: wow.. well that wasn't doing anything 😆

File: test/integration/patients/sidebar/action-sidebar.js [hunk 3]
```diff
@@ -1601,51 +1602,54 @@ context('action sidebar', function() {
...
+      })
       .routeAction(fx => {
+        const testProgramAction = getProgramAction({
+          attributes: {
+            allowed_uploads: ['pdf'],
+          },
+        });
+
         fx.data = getAction({
           id: '1',
```
Comment (paulfalgout) [inline test/integration/patients/sidebar/action-sidebar.js]: can we remove this `id`?

File: test/integration/patients/sidebar/action-sidebar.js [hunk 4]
```diff
@@ -1569,20 +1572,18 @@ context('action sidebar', function() {
...
-        fx.data.attributes.outreach = 'patient';
-        fx.data.attributes.sharing = 'responded';
-        fx.data.relationships['program-action'] = { data: { id: '1' } };
-        fx.data.relationships.form = { data: { id: '11111' } };
-        fx.data.relationships.state = { data: { id: '55555' } };
-        fx.included.push({
-          id: '1',
-          type: 'programs',
+        fx.data = getAction({
+          id: '12345',
```
Comment (paulfalgout) [inline test/integration/patients/sidebar/action-sidebar.js]: would love to remove the non uuid ids from the `getFoo`s in the tests

File: test/integration/patients/sidebar/action-sidebar.js [hunk 5]
```diff
@@ -994,14 +980,14 @@ context('action sidebar', function() {
             },
             meta: {
               view: 'https://www.bucket_name.s3.amazonaws.com/patients/1/view/HRA.pdf',
-              download: 'https://www.bucket_name.s3.amazonaws.com/patients/1/download/HRA.pdf',
+              download: 'https://www.bucket_name.s3.amazonaws.com/patients/$1/download/HRA.pdf',
```
Comment (paulfalgout) [inline test/integration/patients/sidebar/action-sidebar.js]: $1?

**Review by paulfalgout — CHANGES_REQUESTED** [review]
I think with the exception of the specific test fixtures like a clinician we should remove all ids that aren't uuids from the fixture composing and use the generated ones.

File: test/support/api/teams.js (hunk 1 above)
Comment (nmajor25) [reply test/support/api/teams.js]: Yep, switched this 👍

File: test/integration/patients/sidebar/action-sidebar.js (hunk 3 above)
Comment (nmajor25) [reply test/integration/patients/sidebar/action-sidebar.js]: Yep, removed 👍

File: test/integration/patients/sidebar/action-sidebar.js (hunk 4 above)
Comment (nmajor25) [reply test/integration/patients/sidebar/action-sidebar.js]: I removed several more. Think I have all of these removed, besides the ones we need to keep.

File: test/integration/patients/sidebar/action-sidebar.js (hunk 5 above)
Comment (nmajor25) [reply test/integration/patients/sidebar/action-sidebar.js]: Removed that.

**Review by paulfalgout — CHANGES_REQUESTED** [review]
So I think the coverage drop is legitimate from coveralls... so that's.. not going to be fun.

**Comment by nmajor25** [comment]
@paulfalgout I forgot to remove a `.only`, smh 🤦. Hopefully that fixes all the coverage issues.

**Review by paulfalgout — APPROVED** [review]

**Pull request merged**